habitat_baselines/il/trainers/eqa_cnn_pretrain_trainer.py

```diff
@@ -0,0 +1,261 @@
+
+                    avg_loss += loss.item()
+
+                    if t % config.LOG_INTERVAL == 0:
+                        logger.info(
+                            "[ Epoch: {}; iter: {}; loss: {:.3f} ]".format(
+                                epoch, t, loss.item()
+                            )
+                        )
+
+                        writer.add_scalar("total_loss", loss, t)
+                        writer.add_scalars(
+                            "individual_losses",
+                            {"seg_loss": l1, "ae_loss": l2, "depth_loss": l3},
+                            t,
+                        )
+
+                    loss.backward()
+                    optim.step()
+
+                end_time = time.time()
+                time_taken = "{:.1f}".format((end_time - start_time) / 60)
+                avg_loss = avg_loss / len(train_loader)
+
+                logger.info(
+                    "[ Epoch {} completed. Time taken: {} minutes. ]".format(
+                        epoch, time_taken
+                    )
+                )
+                logger.info("[ Average loss: {:.3f} ]".format(avg_loss))
+
+                print("-----------------------------------------")
+
+                if epoch % config.CHECKPOINT_INTERVAL == 0:
+                    self.save_checkpoint(
+                        model.state_dict(), "epoch_{}.ckpt".format(epoch)
+                    )
+
+                epoch += 1
+
+    def _eval_checkpoint(
+        self,
+        checkpoint_path: str,
+        writer: TensorboardWriter,
+        checkpoint_index: int = 0,
+    ) -> None:
+        r"""Evaluates a single checkpoint.
+
+        Args:
+            checkpoint_path: path of checkpoint
+            writer: tensorboard writer object for logging to tensorboard
+            checkpoint_index: index of cur checkpoint for logging
+
+        Returns:
+            None
+        """
+        config = self.config
+
+        config.defrost()
+        config.TASK_CONFIG.DATASET.SPLIT = self.config.EVAL.SPLIT
+        config.freeze()
+
+        env = habitat.Env(config=config.TASK_CONFIG)
+
+        eqa_cnn_pretrain_dataset = EQACNNPretrainDataset(
+            env, config, mode="val"
+        )
+
+        eval_loader = DataLoader(
+            eqa_cnn_pretrain_dataset,
+            batch_size=config.IL.EQACNNPretrain.batch_size,
+            shuffle=False,
+        )
+
+        logger.info(
+            "[ eval_loader has {} samples ]".format(
+                len(eqa_cnn_pretrain_dataset)
+            )
+        )
+
+        model = MultitaskCNN()
+
+        state_dict = torch.load(checkpoint_path)
+        model.load_state_dict(state_dict)
+
+        model.to(self.device).eval()
+
+        depth_loss = torch.nn.SmoothL1Loss()
+        ae_loss = torch.nn.SmoothL1Loss()
+        seg_loss = torch.nn.CrossEntropyLoss()
+
+        np.random.seed(config.EVAL_SEG_COLORS_SEED)
+        self.colors = np.random.randint(255, size=(41, 3))
```

Comment:
Let's use the colors here: https://github.com/facebookresearch/habitat-sim/blob/master/habitat_sim/utils/common.py#L183

Reply:
d069071